pkg/storage/image.go

```diff
@@ -599,16 +600,45 @@ func GetImageService(ctx context.Context, store storage.Store, defaultTransport
 		seenRegistries[r] = true
 	}
 
+	// support global shared /etc/containers/registries.conf
+	allSystemRegistries, err := sysregistriesv2.GetRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+	insecureSystemRegistries := []string{}
+	for _, r := range allSystemRegistries {
+		if r.Insecure {
+			insecureSystemRegistries = append(insecureSystemRegistries, r.URL)
+		}
+	}
+
+	systemRegistries, err := sysregistriesv2.FindUnqualifiedSearchRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+
 	is := &imageService{
 		store:                 store,
 		defaultTransport:      defaultTransport,
 		indexConfigs:          make(map[string]*indexInfo),
 		insecureRegistryCIDRs: make([]*net.IPNet, 0),
-		registries:            cleanRegistries,
 		imageCache:            make(map[string]imageCacheItem),
 		ctx:                   ctx,
 	}
+	if len(cleanRegistries) != 0 {
+		is.registries = cleanRegistries
```

Comment:
(BTW I’m not sure the search behavior can really work, IIRC from working with the Docker backend, the kubelet is unable to provide more than one credential, and it treats unqualified names by using the `docker.io[/library]` normalization when looking up secrets; so there may be no point to this at all. But there may be a different code path involved, not sure.

OTOH, this seems to be the very first code setting `is.registries` to something, so the “search” behavior has never worked anyway. Is it _really_ necessary to introduce it now, considering how it breaks signature security, and all?)

Reply:
chatted offline about this: this is a known issue we do still have in Docker as well, we'll tackle it somehow at some point but this also needs k8s to only provide FQDN which is a no-go at least today to avoid breaking backwards compatibility for them

Reply:
> OTOH, this seems to be the very first code setting is.registries to something, so the “search” behavior has never worked anyway.

This is incorrect — the code was setting `.registries` before.